src/ess_rhat.jl

```diff
@@ -233,25 +236,206 @@ The ESS and ``\\widehat{R}`` values can be computed for the following estimators
 - `StatsBase.mad`
 - `Base.Fix2(Statistics.quantile, p::Real)`
 
+## Types
+
+If no `estimator` is provided, the following `type`s of ESS estimates may be computed:
+- `:bulk`/`:rank`: mean-ESS computed on rank-normalized draws. This type diagnoses poor
```

Comment:
What's the reason for supporting both `:bulk` and `:rank`? Why not just one of them?

Reply:
To simplify both usage and testing of `ess`, `ess_rhat`, and `rhat`. `ess_rhat` and `rhat` both support `type=:rank` and `type=:bulk`. In both cases bulk-ESS is returned, while the R-hat value changes. It makes sense for `ess` to treat `:rank` as a synonym for `:bulk`.

Reply:
Wouldn't it make more sense to only support `:bulk` for `ess`? IMO it is a bit strange to support `:rank` just because `rhat` does and `ess` and `rhat` are somewhat connected via `ess_rhat` which only exists to reduce redundant computations and improve performance.

IMO it would be natural to only support `:bulk` when calling `ess`. And to support `:rank` as well in `ess_rhat` and `rhat`, and document (and implement it such) that in this case `ess` with `:bulk` is returned.

Reply:
I don't feel strongly about this. I'll drop `:rank` from `ess`.